reaper/eeg_reaper.py

```diff
@@ -0,0 +1,187 @@
+            filepaths = [fp for w in os.walk(self.path) for fp in glob.glob(os.path.join(w[0], '*.eeg'))]
+            if not filepaths:
+                raise Warning('no matching files found (or error while checking for files)')
+        except (OSError, Warning) as ex:
+            filepaths = []
+            log.warning(ex)
+        for fp in filepaths:
+            try:
+                eeg = EEGFile(fp, self)
+            except IOError:
+                continue
+            i_state[eeg.reap_id] = reaper.ReaperItem(eeg.reap_state, path=fp)
+        return i_state
+
+    def reap(self, _id, item, tempdir):
+        try:
+            eeg = EEGFile(item['path'], self)
+        except IOError:
+            log.warning('skipping     %s (disappeared or unreadable)', _id)
+            return None, {}
+
+        filepaths = sorted(glob.glob(os.path.splitext(item['path'])[0] + '.*'))
+        filenames = [(fp, os.path.basename(fp)) for fp in filepaths]
+        log.debug('staging      %s%s', _id, ', ' + ', '.join([fn[1] for fn in filenames]))
+        reap_path = os.path.join(tempdir, os.path.basename(item['path']))
+        os.mkdir(reap_path)
+        for fp, fn in filenames:
+            os.symlink(fp, os.path.join(reap_path, fn))
+
+        eeg_size = util.hrsize(item['state']['eeg']['size'])
+        reap_start = datetime.datetime.utcnow()
+        log.info('reaping.zip  %s [%s]', _id, eeg_size)
+        try:
+            filepath = util.create_archive(reap_path, os.path.basename(reap_path), rootdir=False)
+            shutil.rmtree(reap_path)
+        # pylint: disable=broad-except
+        except Exception:
+            log.warning('reap error   %s', _id)
+            return False, None
+        metadata = util.object_metadata(eeg, self.timezone, os.path.basename(filepath))
+        metadata['acquisition']['files'][0]['info'] = {'path': item['path']}
+        util.set_archive_metadata(filepath, metadata)
+        reap_time = (datetime.datetime.utcnow() - reap_start).total_seconds()
+        log.info('reaped.zip   %s [%s] in %.1fs', _id, eeg_size, reap_time)
+        return True, {filepath: metadata}
+
+
+class EEGFile(object):
+
+    """EEGFile class"""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, filepath, reaper_inst):
+        # check that file exists and is readable
+        with open(filepath, 'rb'):
+            self.filepath = filepath
+
+        # the .eeg file's ctime changes when appended, using header's ctime instead
+        header = os.path.splitext(filepath)[0] + '.vhdr'
+        creation_time = os.stat(header).st_ctime
+
+        relpath = os.path.relpath(filepath, reaper_inst.path)
+        dirpath, filename = os.path.split(relpath)
+        hierarchy_info = dirpath.split('/') if dirpath else []
+        filename_info = os.path.splitext(filename)[0].split('_')
+        sort_info = hierarchy_info + list(filter(None, filename_info))
```

Comment:
I think the `list()` call is superfluous here. And, generally, `filter()`, `map()`, and `reduce()` should be avoided in favor of more readable / pythonic list comprehensions. I'm surprised pylint isn't complaining about this.

`[fni for fni in filename_info if fni]`